Reflection/primitive_types.dsl.h

```diff
@@ -1,7 +1,29 @@
+// This file is used for mass registering of primitives type handlers in reflection and serialization routines.
+// Typical usecase:
+//    #define CURRENT_DECLARE_PRIMITIVE_TYPE(typeid_index, cpp_type, current_type) \
```

Comment:
:+1: 

Don't see a good reason to not put the license header into this file too.